datalad/distribution/tests/test_siblings.py

```diff
@@ -0,0 +1,175 @@
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Test adding sibling(s) to a dataset
+
+"""
+
+from os.path import join as opj, basename
+from datalad.api import install, siblings
+from datalad.support.gitrepo import GitRepo
+from datalad.support.exceptions import InsufficientArgumentsError
+
+from datalad.tests.utils import chpwd
+from datalad.tests.utils import with_tempfile, with_testrepos
+from datalad.tests.utils import assert_in
+from datalad.tests.utils import assert_not_in
+from datalad.tests.utils import assert_raises
+from datalad.tests.utils import assert_status
+from datalad.tests.utils import assert_result_count
+
+from nose.tools import eq_, ok_
+
+
+# work on cloned repos to be safer
+@with_testrepos('submodule_annex', flavors=['clone'])
+@with_tempfile(mkdir=True)
+def test_siblings(origin, repo_path):
+
+    sshurl = "ssh://push-remote.example.com"
+    httpurl1 = "http://remote1.example.com/location"
+    httpurl2 = "http://remote2.example.com/location"
+
+    # insufficient arguments
+    # we need a dataset to work at
+    with chpwd(repo_path):  # not yet there
+        assert_raises(InsufficientArgumentsError,
+                      siblings, 'add', url=httpurl1)
+
+    # prepare src
+    source = install(repo_path, source=origin, recursive=True)
+    # pollute config
+    depvar = 'remote.test-remote.datalad-publish-depends'
+    source.config.add(depvar, 'stupid', where='local')
+
+    # cannot configure unknown remotes as dependencies
+    res = siblings(
+        'configure',
+        dataset=source,
+        name="test-remote",
+        url=httpurl1,
+        publish_depends=['r1', 'r2'],
+        on_failure='ignore')
+    assert_status('error', res)
+    assert_in('unknown sibling(s) specified as publication dependency',
+              res[0]['message'])
+    # prior config was changed by failed call above
+    eq_(source.config.get(depvar, None), 'stupid')
+
+    res = siblings('configure',
+                   dataset=source, name="test-remote",
+                   url=httpurl1,
+                   result_xfm='paths')
+
+    eq_(res, [source.path])
+    assert_in("test-remote", source.repo.get_remotes())
+    eq_(httpurl1,
+        source.repo.get_remote_url("test-remote"))
+
+    # reconfiguring doesn't change anything
+    siblings('configure', dataset=source, name="test-remote",
+             url=httpurl1)
+    assert_in("test-remote", source.repo.get_remotes())
+    eq_(httpurl1,
+        source.repo.get_remote_url("test-remote"))
```

Comment:
May be also test config for not polluting it with redundant entries? Theoretically reconfiguring could result in several `remote.test-remote.url` entries, that are identical. This might not lead to failure of these assertions but would still be annoying.